tests/src/system/basic/WskBasicTests.scala

```diff
@@ -333,6 +333,26 @@ class WskBasicTests
             }
     }
 
+    /**
+      * Tests creating an nodejs action that throws a whisk.error() response. The error message thrown by the
+      * whisk.error() should be returned.
+      */
+    it should "create and invoke a blocking action resulting in a whisk.error response" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "whiskError"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getTestActionFilename("applicationError1.js")))
+            }
+
+            if (usePythonCLI) {
+                wsk.action.invoke(name, blocking = true, expectedExitCode = 246)
+                  .stderr should include regex (""""error": "This error thrown on purpose by the action."""")
+            } else {
+                wsk.action.invoke(name, blocking = true)
```

Comment:
When retrieving the activation information for an invocation that returns a whisk.error() response via "activation get," zero is returned as an exit code. However, when invoking a blocking action that returns a whisk.error() response with "action invoke --blocking," a non-zero exit code is returned. This logic quite strange. Retrieve an activation record should return the same exit code as an invocation.

Also, the way the errors are handled in the Go CLI makes in more difficult to return JSON responses with errors.